File: mobile/spag_app/lib/features/technician/models/installation_job.dart
class InstallationJob {
  final int requestId;
  final int? serviceId;
  final String customerName;
  final String customerPhone;
  final String address;
  final String modelName;
  final int purifierModelId;
  final String status;

  InstallationJob({
    required this.requestId,
    this.serviceId,
    required this.customerName,
    required this.customerPhone,
    required this.address,
    required this.modelName,
    required this.purifierModelId,
    required this.status,
  });

  factory InstallationJob.fromJson(Map<String, dynamic> json) {
    return InstallationJob(
      requestId: json['request_id'] as int? ?? 0,
      customerName: json['customer_name'] as String? ?? '',
      customerPhone: json['customer_phone'] as String? ?? '',
      address: json['address'] as String? ?? '',
      modelName: json['model_name'] as String? ?? 'Unknown',
      purifierModelId: json['purifier_model_id'] as int? ?? 0,
      status: json['status'] as String? ?? 'unknown',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'request_id': requestId,
      'service_id': serviceId,
      'customer_name': customerName,
      'customer_phone': customerPhone,
      'address': address,
      'model_name': modelName,
      'purifier_model_id': purifierModelId,
      'status': status,
    };
  }
}

File: backend/app/schemas/installation_job.py
from pydantic import BaseModel
from typing import Optional
from datetime import date

class InstallationJobResponse(BaseModel):
    """Schema for installation jobs returned to technician"""
    
    request_id: int
    customer_name: str
    customer_phone: str
    address: str
    model_name: str
    status: str
    purifier_model_id: Optional[int] = None

    class Config:
        from_attributes = True


class CompleteInstallationRequest(BaseModel):
    """Schema for completing installation with required fields"""
    customer_name: str
    address: str
    installation_date: date
    site_details: str
    purifier_model_id: int
    address_line2: Optional[str] = None
    city: Optional[str] = None
    state: Optional[str] = None
    pincode: Optional[str] = None
    landmark: Optional[str] = None
    mobile_number: Optional[str] = None
    email: Optional[str] = None
